src/v/cluster/feature_manager.cc

```diff
@@ -0,0 +1,267 @@
+/*
+ * Copyright 2021 Vectorized, Inc.
+ *
+ * Use of this software is governed by the Business Source License
+ * included in the file licenses/BSL.md
+ *
+ * As of the Change Date specified in that file, in accordance with
+ * the Business Source License, use of this software will be governed
+ * by the Apache License, Version 2.0
+ */
+
+#include "feature_manager.h"
+
+#include "cluster/cluster_utils.h"
+#include "cluster/commands.h"
+#include "cluster/health_monitor_frontend.h"
+#include "cluster/members_table.h"
+#include "feature_table.h"
+#include "raft/group_manager.h"
+
+namespace cluster {
+
+static constexpr std::chrono::seconds status_retry = 5s;
+
+ss::future<> feature_manager::start() {
+    vlog(clusterlog.info, "Starting...");
+    _leader_notify_handle
+      = _group_manager.local().register_leadership_notification(
+        [this](
+          raft::group_id group,
+          model::term_id term,
+          std::optional<model::node_id> leader_id) {
+            // Should never be called with gate closed: group manager
+            // is shut down: we unregister leader notification before
+            // closing gate.
+            _gate.check();
+
+            if (group != _raft0_group) {
+                return;
+            }
+
+            vlog(
+              clusterlog.info, "Controller leader notification term {}", term);
+
+            // This hook avoids the need for the controller leader to receive
+            // its own health report to generate a call to update_node_version.
+            // Especially useful on first startup of cluster, where the initial
+            // leader is the only node, and can immediately store the cluster
+            // version based on its own version alone.
+            if (
+              _feature_table.local().get_active_version() != latest_version
+              && leader_id.has_value()
+              && leader_id.value() == config::node().node_id()) {
```

Comment:
`leader_id == config::node().node_id()` should be enough, `std::optional` has a special `==` operator override

Reply:
done.